lightningd/pay.c

```diff
@@ -1487,16 +1495,21 @@ static struct command_result *json_waitsendpay(struct command *cmd,
 	struct sha256 *rhash;
 	unsigned int *timeout;
 	struct command_result *res;
-	u64 *partid;
+	u64 *partid, *groupid;
 
 	if (!param(cmd, buffer, params,
 		   p_req("payment_hash", param_sha256, &rhash),
 		   p_opt("timeout", param_number, &timeout),
 		   p_opt_def("partid", param_u64, &partid, 0),
+		   p_opt("groupid", param_u64, &groupid),
 		   NULL))
 		return command_param_failed();
 
-	res = wait_payment(cmd->ld, cmd, rhash, *partid);
+	if (groupid == NULL) {
+		groupid = tal(cmd, u64);
+		*groupid = wallet_payment_get_groupid(cmd->ld->wallet, rhash);
```

Comment:
Would prefer this function to be called "wallet_payment_get_last_groupid" to be clearer? (or max_groupid)?